Issue 19 python data analytics

Merwanski opened the pull request.

Merwanski committed:
added last n data function to the ventilator databse

Merwanski committed:
added function last n values

Merwanski committed:
database processing class for the pressure values

frankvanbever commented on ventilator_database.py:
```diff
@@ -29,30 +29,60 @@ def store_trigger(self, trigger_val):
+    def last_n_data(self, type_data, N=100):
+        # retrieve the last "N" added measurement 
+        if type_data == 'BPM':
+            return self.db.breathsperminute_values.find().sort("loggedAt", -1).limit(N)
```
Wouldn't it make more sense to keep a circular buffer of these values locally, rather than having to go out to the database latency wise?

frankvanbever commented on database_processing.py:
```diff
@@ -0,0 +1,19 @@
+        super().__init__()
+    # TODO
+    def run(self):
+        return -1
```
It's complaining here about lack of newline

frankvanbever reviewed (changes requested):
I'm wondering whether it makes sense to go out to the database to get these values instead of keeping the values in a circular buffer on the incoming path? I wasn't here during the day so I'm unaware of design decisions that went into this. @TomBruyneel you know more about Mongo than I do, wdyt?

There's also some trailing whitespace that will create noise in git blame, if you could remove those changes that'll reduce the noise a bit.

Merwanski committed:
Issue #29 function to compute the effective BPM value from the loaded dataset

Merwanski committed:
updated the class definition and  function get nbr bpm finalized

Merwanski committed:
added function to analyze inhale exhale timestamp

the function returns the following
nbr_ratio_below_threshold (Te/Ti > threshold ~3 )
nbr_dtinhale_above_threshold (Ti > 10)
nbr_dtexhale_above_threshold (Te > 10)

Merwanski committed:
check_pressure_tracking_performance

function added to the python_daemon/database_processing.py
The function will compare the last n measure of the inhale phase and compare it to the desired pressure set and return the following:
* nbr_dp_above_threshold: how many time dp was above the threshold set
* dp_list : all the average  values of the differences computed between the measured pressure and the desired pressure over all the breathing cycles

Merwanski committed:
detect_pressure_below_peep function added

function added to the python_daemon/database_processing.py
detect_pressure_below_peep(self, peep_value = 10, threshold_dp_peep = 5, nbr_data_point = 35)
the function returns the following

nbr_dp_peep_above_threshold on the analyzed breathing cycles
below_peep_list all the values detected

Merwanski committed:
pressure_peak_overshoot function added

pressure_peak_overshoot(self, pressure_desired = 51, threshold_dp_overshoot = 3, nbr_data_point = 10):
and return the following
nbr_pressure_overshoot_above_threshold,
overshoot_pressure_list
